### PR TITLE
Override FlushAsync in NLogTarget to flush contained TelemetryClient

### PR DESCRIPTION
**Pull request by nescientguru**
# Overview
This PR implements the `FlushAsync` method in `ApplicationInsightsTarget` to utilize the preferred NLog mechanism to flush the contained `TelemetryClient`.

## Adapter Changes
- Implement `ApplicationInsightsTarget.FlushAsync` by flushing `TelemetryClient` and calling base class 'FlushAsync'

## Unit Test Changes
- Implement `Flush` method in `CustomTelemetryChannel`
- Expose `FlushCount` property on `CustomTelemetryChannel`
- Initialize and cleanup `FlushCount` as appropriate
- Create `TargetFlushesTelemetryClient` unit test to verify proper execution of `ApplicationInsightsTarget.FlushAsync`

### CODE REVIEW

**Comment by snakefoot** [comment]
@nescientguru Cannot see that Flush does anything, besides increasing a counter, that you then validate in the unit-test. What is the purpose of this PR?

File: src/Adapters/NLogTarget.Shared/ApplicationInsightsTarget.cs [hunk 1]
```diff
@@ -50,6 +51,16 @@ internal TelemetryClient TelemetryClient
         }
 
         /// <summary>
+        /// Flush any pending log messages asynchronously (in case of asynchronous targets).
+        /// </summary>
+        /// <param name="asyncContinuation">The asynchronous continuation.</param>
+        protected override void FlushAsync(AsyncContinuation asyncContinuation)
+        {
+            this.telemetryClient.Flush();
```
Comment (SergeyKanzhelev) [inline src/Adapters/NLogTarget.Shared/ApplicationInsightsTarget.cs]: is it expected to block the thread by `telemetryClient.Flush()` inside the `FlushAsync` call? Maybe it should be wrapped into the `new Task(() => {tc.Flush()} ).Start()` and hook it up into the asyncContinuation?

**Comment by SergeyKanzhelev** [comment]
Would you mind write a simple unit test for this one so two adaptors will be covered in one shot: https://github.com/Microsoft/ApplicationInsights-dotnet-logging/pull/69

**Comment by MS-TimothyMothra** [comment]
This PR has merge conflicts and requires additional work.
I'm closing this and creating a new issue #155 to track this as a future work item.

**Pull request closed without merging**